model/src/main/java/com/epam/esm/repository/TagRepository.java

```diff
@@ -0,0 +1,67 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.entities.Tag;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * An interface {@code TagRepository} defines the methods for retrieving tag data from the database;
+ *
+ * @author Sviatlana Shelestava
+ * @see com.epam.esm.repository.TagRepositoryImpl
```

Comment:
Incorrect direction. You can link interface in implementation of needed but not vise versa.